spec/lib/wolf_waagen_api/import_spec.rb

```diff
@@ -0,0 +1,250 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe WolfWaagenApi::Import do
+  let(:report) do
+    Report.create!(
+      name: 'Example Report',
+      hive_id: 'HIVE_ID',
+      start_date: DateTime.new(2019, 1, 1)
+    )
+  end
+
+  let(:other_report) do
+    Report.create!(
+      name: 'Other Report',
+      hive_id: 'HIVE_ID2'
+    )
+  end
+
+  let(:wolf_sensor_type) do
+    specified_type = WolfWaagenApi::Import::SENSOR_TYPES.first
+
+    SensorType.create!(
+      property: specified_type[:property],
+      unit: specified_type[:unit]
+    )
+  end
+
+  let(:wolf_sensor) do
+    Sensor.create!(
+      sensor_type: wolf_sensor_type,
+      name: 'Wolf Sensor 1',
+      report: report,
+      address: 1
+    )
+  end
+
+  let(:wolf_sensor2) do
+    Sensor.create!(
+      sensor_type: wolf_sensor_type,
+      name: 'Wolf Sensor 2',
+      report: report,
+      address: 2
+    )
+  end
+
+  let(:wolf_sensor_reading) do
+    Sensor::Reading.create!(
+      sensor: wolf_sensor,
+      created_at: DateTime.new(2019, 1, 2),
+      calibrated_value: 100,
+      uncalibrated_value: 100
+    )
+  end
+
+  let(:wolf_sensor_reading2) do
+    Sensor::Reading.create!(
+      sensor: wolf_sensor2,
+      created_at: DateTime.new(2019, 1, 3),
+      calibrated_value: 100,
+      uncalibrated_value: 100
+    )
+  end
+
+  let(:other_sensor_type) do
+    SensorType.create!(
+      property: 'Other Sensor Type',
+      unit: ''
+    )
+  end
+
+  let(:other_type_sensor) do
+    Sensor.create!(
+      sensor_type: other_sensor_type,
+      name: 'Other Sensor',
+      report: report,
+      address: 3
+    )
+  end
+
+  let(:other_report_sensor) do
+    Sensor.create!(
+      sensor_type: wolf_sensor_type,
+      name: 'Other Report’s Wolf Sensor',
+      report: other_report,
+      address: 4
+    )
+  end
+
+  describe '#initialize' do
+    it 'validates report' do
+      expect do
+        described_class.new(report: nil)
+      end.to raise_error ArgumentError
+    end
+
+    it 'ensures report has a `hive_id`' do
+      expect do
+        described_class.new(report: Report.first)
+      end.to raise_error ArgumentError
+    end
+  end
+
+  describe '#api_sensors' do
+    subject do
+      described_class.new(report: report).api_sensors
+    end
+
+    context 'given sensors of types not sepcified in constant' do
+      before { other_type_sensor }
+
+      it {
+        is_expected.not_to include(other_type_sensor)
+      }
+    end
+
+    context 'given sensor for other reports' do
+      before { other_type_sensor }
+
+      it { is_expected.not_to include(other_report_sensor) }
+    end
+
+    context 'given no matching sensor' do
+      before { other_sensor_type }
+
+      it { is_expected.to eq([]) }
```

Comment:
Tests that check on non-existence are very prone to false negatives. The reason is that you never know *why* sth. is *not* there. It is better to check for existence.

If you must, there is a pattern where you create `A` and `B` which are very similar except for a particular property. In the test you can check that only `A` exist or is included. In that case it is obvious to the reader why sth. is there and why not *and* the whole test is robust against false negatives.